recipe/2024-01-support-private-terraform-repository.md

```diff
@@ -0,0 +1,237 @@
+# Adding support for Terraform Modules from Private Git Repository
+
+* **Status**: Pending
+* **Author**: Vishwanath Hiremath (@vishwahiremat)
+
+## Overview
+
+Today, radius can work with publicly-hosted Terraform modules across a few different module sources. We don't support working with privately-hosted Terraform modules, and there's no way to configure authentication.
+
+This is important because organizations write their own Terraform modules and store them in privately-accessible sources. In order for us to support serious use, we need to enable private sources.
+
+## Terms and definitions
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Terraform module source | The source argument in a module block tells Terraform where to find the source code for the desired child module |
+| Module registry | A module registry is the native way of distributing Terraform modules for use across multiple configurations |
+| Terraform registry | Terraform Registry is an index of modules shared publicly |
+| HTTP URLs | When you use an HTTP or HTTPS URL, Terraform will make a GET request to the given URL, which can return another source address | 
+| Private Terraform Repository | A private Terraform repository typically refers to a version control repository that contains Terraform module code, but is not publicly accessible. | 
+
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6911
+
+### Goals
+- Enable support to use terraform modules from private Git repositories.
+- It should support git from any platforms(like Bitbucket, Gitlab, Azure DevOps Git etc.)
+
+### Non goals
+- To Support other terraform module sources like S3, GCP, Mercurial repository.
+- Registering Terraform modules stored in repositories from different git accounts.
```

Comment:
Not sure if I understand this.. do you mean only supporting one git account per environment?

Reply:
Can you please update this non-goal to be more explicit.. support for multiple git accounts per Radius environment?